utilities/random_number_generation/bbs.py

```diff
@@ -0,0 +1,42 @@
+from primes import getStrongBlumPrime
+
+"""
+    An implementation of the Blum Blum Shub pseduo-random 
+    number generator.
+    
+    General Notes:
+        - BBS must be initialized with a random seed
+        - In order for BBS to produce cryptographically secure
+          output, the seed must be truly random
+        - In order for the generator to be practically secure, the 
+          bit length of the modulus should be no less than 2048
+        - The functions used to generate the modulus of the
+          function rely on PyCryptodome's Crypto.Util package
```

Comment:
Is this something we should add to the requirements file? And would this be something we could eventually do ourselves? Or is it unnecessary?

Reply:
Yes, yes, and yes. We should add it to the requirements file. I think the package name was just "pycryptodome". We could do it ourselves but it's somewhat unnecessary. I'm not sure exactly how pycryptodome generates primes but it probably uses the Miller-Rabin Primality Test to test random numbers for primality. We don't really need to do this though because pycryptodome does it for us. One might wonder if it's "truly" secure since the random primes aren't dependent on quantum randomness but the quantum randomness of the seed will be enough. We also don't want to implement too much of the classical-crypto ourselves because it increases the change of security flaws from bugs so if we don't need to manually modify part of the algorithm, it's best just to use some well-tested library.

Reply:
Sounds good. I agree that doing too much of that ourselves both introduces security flaws and may also be a bit unnecessary and more of a fun extra thing. Thank you for going into depth about it.